### ISSUE TITLE
protoc-gen-go: print help when invoked directly

### ISSUE THREAD

**Issue by atombender** [golang/protobuf#239]
Flags such as `-M` and `--plugins` seem to be entirely undocumented. I expected to find them [here](https://developers.google.com/protocol-buffers/docs/reference/go-generated) or at least be able to run `protoc-gen-go --help`.

**Comment by atombender** [comment]
Finally found them buried in [this](https://github.com/golang/protobuf#parameters) section.

**Comment by Omustardo** [comment]
+1 to this. The use of nested flags to set values with plugins such as:
`protoc --go_out=plugins=grpc,import_path=mypackage:. *.proto`
is rather confusing. 

From what I understand, `--go_out` is a flag that tells `protoc` to use the `protoc-gen-go` plugin to compile a proto to golang. Since the `protoc-gen-go` plugin isn't being called directly it needs to get flags passed in as nested flags. In this case, it's like calling `protoc-gen-go` with the flags:
`--plugins=grpc --import_path=mypackage`

The flags are parsed in protoc-gen-go/generator/generator.go

**Comment by fuzzyami** [comment]
these nested flags are unfortunate. They raise the learning curve needlessly.

**Comment by atombender** [comment]
Perhaps this project could consider the "subargs" strategy used by [Browserify](http://browserify.org). It would look something like this:

```
protoc --go_out [ --plugins=grpc --import_path=mypackage:.* ] *.proto
```

In other words, whatever is inside the brackets are simply normal flags, but they only apply to `go_out`.

**Comment by bweston92** [comment]
I totally agree, I've been going round endlessly.

**Comment by neild** [comment]
The nested flags are unfortunate, but follow the convention for other protoc output modes. Consistency is better than inventing our own thing here.

They're all documented in README.md, FWIW:
https://github.com/golang/protobuf#parameters

Having protoc-gen-go print something when invoked directly seems reasonable, although I'm not certain how to detect when that's happened short of vendoring in something like https://github.com/mattn/go-isatty.

**Comment by dsnet** [comment]
When a person encounters a binary that they know nothing about, it's pretty customary to simply invoke the `--help` flag.

Thus, perhaps calling `protoc-gen-go --help` should print the help and exit.

**Comment by dsnet** [comment]
This is fixed in the [v1.20.0 release](https://github.com/protocolbuffers/protobuf-go/releases/tag/v1.20.0).

**Comment by ashutoshkumr** [comment]
@dsnet could you please confirm what was fixed as part of this or if the steps for checking help is different from what's described above ?

I'm basically trying to list all supported options (more specifically usage of M flag) but this didn't work !
```sh
# curl -LkO https://github.com/protocolbuffers/protobuf-go/releases/download/v1.20.0/protoc-gen-go.v1.20.0.linux.amd64.tar.gz
# tar xzvf protoc-gen-go.v1.20.0.linux.amd64.tar.gz
# ./protoc-gen-go --help
protoc-gen-go: unknown argument "--help" (this program should be run by protoc, not directly)
```
